### DISCUSSION TITLE
Per-asset configuration: query parameters or something else?

### DISCUSSION THREAD

**Discussion by rhendric** [parcel-bundler/parcel#5214] [Q&A]
I'm trying to write a Parcel v2 transformer that edits font files down to a given subset of glyphs.

Each font file should have its glyph subset defined independently, and I also want to be able to embed the resulting font directly into the CSS referencing it with a `data:` URL. My first thought was to use query parameters in the URL that Parcel resolves: something like `url('path/to/source/font.woff2?subset=<glyphs to include>')`, with an optional `data-url:` at the front if I want to embed the resulting font. However, starting down this path, my `Transformer` gets called but the `query` property on the asset is undefined.

Is there another way to get the query parameters of an asset from a transformer? Or, perhaps more importantly, does Parcel want me to provide per-asset configuration like this in some other way, a way which hopefully is also compatible with using a named pipeline like `data-url:`?

**Comment by mischnic** [comment]
This already exists: https://github.com/parcel-bundler/parcel/blob/069a2b1d773cb9fe02d621c2a281ecb47b5bf668/packages/transformers/image/src/ImageTransformer.js#L19-L24

```js
import image from 'url:./image.jpg?height=200';
import image from 'url:./image.webp?height=200&width=300';
import image from 'url:./image.png?width=300';
```

https://github.com/parcel-bundler/parcel/pull/4881

**Reply by rhendric** [reply]
Yeah, that was my initial inspiration; I just didn't know if it was the latest encouraged pattern, since I could only find it used in that one transformer.

Turns out the reason my attempts weren't working is down to a bug that will hopefully be fixed when #5228 is merged.

**Reply by mischnic** [reply]
> if it was the latest encouraged pattern, since I could only find it used in that one transformer.

It's just that it wasn't needed for any other transformer previously and it was then implemented together with the image transformer.
